src/aws-elasticache/serverless-cache.ts

```diff
@@ -0,0 +1,534 @@
+    return aws_iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        Stack.of(this).formatArn({
+          service: 'elasticache',
+          resource: 'serverlesscache',
+          resourceName: this.serverlessCacheName,
+          arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+        }),
+      ],
+    });
+  }
+
+  /**
+   * Permits an IAM principal to perform connect to the serverless cache.
+   *
+   * Actions: Connect
+   *
+   * @param grantee The principal to grant access to.
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/auth-iam.html
+   */
+  public grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant {
+    return this.grant(grantee, 'elasticache:Connect');
+  }
+
+  /**
+   * Create a CloudWatch metric for serverless cache.
+   *
+   * @param metricName name of the metric.
+   * @param props metric options.
+   *
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/serverless-metrics-events-redis.html
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/serverless-metrics-events.memcached.html
+   */
+  public metric(metricName: string, props?: MetricOptions): Metric {
+    return new Metric({
+      namespace: 'AWS/ElastiCache',
+      dimensionsMap: {
+        CacheClusterId: this.serverlessCacheName,
+      },
+      metricName,
+      ...props,
+    });
+  }
+
+  /**
+   * Metric for the total number of bytes used by the data stored in your cache.
+   *
+   * @default - average over 5 minutes
+   */
+  public metricBytesUsedForCache(props?: MetricOptions): Metric {
+    return this.metric('BytesUsedForCache', { statistic: 'Average', ...props });
+  }
+
+  /**
+   * Metric for the total number of ElastiCacheProcessingUnits (ECPUs) consumed by the requests executed on your cache.
+   *
+   * @default - average over 5 minutes
+   */
+  public metricElastiCacheProcessingUnits(props?: MetricOptions): Metric {
+    return this.metric('ElastiCacheProcessingUnits', { statistic: 'Average', ...props });
+  }
+}
+
+/**
+ * Represents a ElastiCache Serverless Cache construct in AWS CDK.
+ *
+ * @example
+ * declare const vpc: aws_ec2.IVpc;
+ *
+ * const serverlessCache = new ServerlessCache(
+ *   stack,
+ *   'ServerlessCache',
+ *   {
+ *     serverlessCacheName: 'my-serverlessCache',
+ *     engine: Engine.VALKEY,
+ *     vpc,
+ *   },
+ * );
+ */
+export class ServerlessCache extends ServerlessCacheBase implements IServerlessCache {
+  /**
+   * The serverless cache ARN.
+   */
+  readonly serverlessCacheArn: string;
+  /**
+   * The serverless cache name.
+   */
+  readonly serverlessCacheName: string;
+  /**
+   * The DNS hostname of the cache node.
+   */
+  readonly endpointAddress: string;
+  /**
+   * The port number that the cache engine is listening on.
+   */
+  readonly endpointPort: number;
+
+  /**
+   * The connection object associated with the ElastiCache Serverless Cache.
+   */
+  public readonly connections: aws_ec2.Connections;
```

Comment:
according to the `connections`:

```suggestion
  /**
   * The serverless cache ARN.
   */
  public readonly serverlessCacheArn: string;
  /**
   * The serverless cache name.
   */
  public readonly serverlessCacheName: string;
  /**
   * The DNS hostname of the cache node.
   */
  public readonly endpointAddress: string;
  /**
   * The port number that the cache engine is listening on.
   */
  public readonly endpointPort: number;

  /**
   * The connection object associated with the ElastiCache Serverless Cache.
   */
  public readonly connections: aws_ec2.Connections;
```